parts/kubernetesmastercustomscript.sh

```diff
@@ -249,6 +249,23 @@ function ensureEtcd() {
     done
 }
 
+function ensureEtcdDataDir() {
+    etcdStarted=1
+    mount | grep /dev/sdc1 | grep /var/lib/etcddisk
+    if [ "$?" = "0" ]
+    then
+        echo "Etcd running with data dir at: /var/lib/etcddisk"
+        etcdStarted=0
+        break
+    fi
+
+    if [ $etcdStarted -ne 0 ]
+    then
+        echo "Etcd data dir was not found at: /var/lib/etcddisk"
```

Comment:
I consider mounting a basic operation and should not have to write code to retry it IMO. DC/OS also creates fs and mounts it in cloud init (ephemeral).

I don't think there are any retries built into cloud init's syntactical way of specifying mount and fs setup. If you want a retry then etcd disk setup needs to be move to CSE which is not bad by itself but I am not convinced that retries need to build for something as fundamental as mounting.

Running tests in a loop for a couple of days will give me better idea about mount reliability.